### DISCUSSION TITLE
`group` with custom variants does not accept modifiers.

### DISCUSSION THREAD

**Discussion by fenilli** [tailwindlabs/tailwindcss#9788] [Help] [answered]
When creating a new variant, you would also want to have it work with `group`.

This works fine to add groups
`addVariant("group-formkit-invalid", "[data-invalid='true']:merge(.group) &")`

But if we want to use the modifiers like `group/item` for nested groups, there isn't documentation on how to add, and looking at the code that generates that it uses `matchVariant` a function that is internal and not exposed to the plugin.

So how to make `group/item` and `group-formkit-invalid/item` work?

**Comment by wongjn** [comment, marked as answer]
`matchVariants()` is not an exposed interface layer and I couldn't find any core code that has this or uses this function, but [`matchVariant()` *is* used by core and exposed to plugins](https://tailwindcss.com/blog/tailwindcss-v3-2#dynamic-variants-with-matchvariant) and you can use this to adapt the [core implementation](https://github.com/tailwindlabs/tailwindcss/blob/88dcb6ebc51093ab2289dd7405c6c915d863ad71/src/corePlugins.js#L151-L170) to work for your case:
```js
module.exports = {
  plugins: [
    require('tailwindcss/plugin')(({ matchVariant }) => {
      matchVariant(
        'group',
        (value= '', { modifier }) =>
          modifier
            ? `[data-invalid='true']:merge(.group\\/${modifier}) &`
            : `[data-invalid='true']:merge(.group) &`,
        { values: { 'formkit-invalid': 'formkit-invalid' } },
      );
    })
  ],
}
```
See it at work in [this Tailwind Play](https://play.tailwindcss.com/JSu4IS72pj?file=config) (open the *Generated CSS* panel to see the generated utilities).

**Reply by fenilli** [reply]
Yeh my mistake about `matchVariants` typo, oh so it is exposed, for some reason when doing `{ matchVariant }` inside a plugin using TS it does not find any types for matchVariant.

Thanks for the repo and example.
